mqtt_client.c

```diff
@@ -1139,9 +1123,17 @@ static int mqtt_message_receive(esp_mqtt_client_handle_t client, int read_poll_t
          * type and flags.
          */
         read_len = esp_transport_read(t, (char *)buf, 1, read_poll_timeout_ms);
-        if (read_len <= 0) {
-            return esp_mqtt_handle_transport_read_error(read_len, client);
+        if (read_len == 0) {
+            // On ESP-IDF v5+ and v4.4, esp_transport_read()=0 means timeout.
+            ESP_LOGD(TAG, "mqtt_message_receive(): no message (read timeout)");
+            return 0;
         }
+        if (read_len < 0) {
+            // On ESP-IDF v5+ and v4.4, esp_transport_read()<0 on close or error.
+            esp_mqtt_client_dispatch_transport_error(client);
+            return -1;
+        }
+
```

Comment:
Same here. Please revert

Reply:
Changed to a bit of logic around `esp_mqtt_handle_transport_read_error()` as discussed above.